File: src/main/java/coffee/amo/interactionkey/api/InteractableBlockEntity.java
package coffee.amo.interactionkey.api;

import net.minecraft.core.BlockPos;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.level.block.entity.BlockEntity;
import net.minecraft.world.level.block.entity.BlockEntityType;
import net.minecraft.world.level.block.state.BlockState;

// Does nothing until i figure out what to do with it.
public class InteractableBlockEntity extends BlockEntity implements Interactable {
    public BlockEntity blockEntity;

    public InteractableBlockEntity(BlockEntityType<?> pType, BlockPos pPos, BlockState pBlockState) {
        super(pType, pPos, pBlockState);
    }

    public void interact() {

    }

    @Override
    public String getInteractableName() {
        return this.blockEntity.getLevel().getBlockState(this.blockEntity.getBlockPos()).getBlock().getName().getString();
    }

    @Override
    public String getDescription() {
        return null;
    }

    @Override
    public int getPriority() {
        return 0;
    }

    @Override
    public void interact(Player player) {

    }
}

File: src/main/kotlin/coffee/amo/interactionkey/api/InteractableBlockEntity.kt
package coffee.amo.interactionkey.api

import net.minecraft.core.BlockPos
import net.minecraft.world.entity.player.Player
import net.minecraft.world.level.block.entity.BlockEntity
import net.minecraft.world.level.block.entity.BlockEntityType
import net.minecraft.world.level.block.state.BlockState

// Does nothing until i figure out what to do with it.
class InteractableBlockEntity(pType: BlockEntityType<*>?, pPos: BlockPos?, pBlockState: BlockState?) : BlockEntity(pType, pPos, pBlockState), Interactable {
    var blockEntity: BlockEntity? = null
    fun interact() {}
    override val interactableName: String
        get() = blockEntity!!.level!!.getBlockState(blockEntity!!.blockPos).block.name.string
    override val description: String?
        get() = null
    override val priority: Int
        get() = 0

    override fun interact(player: Player?) {}
}